receiver/snmpreceiver/config.go

```diff
@@ -448,18 +455,28 @@ func validateColumnOID(metricName string, columnOID ColumnOID, cfg *Config) erro
 
 	// Check that any ResourceAttributes have a valid value
 	if len(columnOID.ResourceAttributes) > 0 {
-		hasIndexedIdentifier = true
```

Comment:
If we are getting rid of this line, we don't need the if condition above either. An empty list will be skipped by the for loop.